Sources/NIO/EventLoop.swift

```diff
@@ -59,10 +59,12 @@ public final class RepeatedTask {
     private var scheduled: Scheduled<EventLoopFuture<Void>>?
     private var task: (() -> EventLoopFuture<Void>)?
 
-    internal init(interval: TimeAmount, eventLoop: EventLoop, task: @escaping () -> EventLoopFuture<Void>) {
+    internal init(interval: TimeAmount, eventLoop: EventLoop, task: @escaping (RepeatedTask?) -> EventLoopFuture<Void>) {
         self.delay = interval
         self.eventLoop = eventLoop
-        self.task = task
+        self.task = { [weak self] in
```

Comment:
You can avoid the weak reference by recreating the task each time. That is, you call `scheduleTask(in: delay) { task(self) }`.

That does provide a reference cycle, but it's one that already exists, as `reschedule` already gives the `Future` a reference to `self`. Then there's no need to receive the `RepeatedTask` as an optional.